### DISCUSSION TITLE
USER_BACKLOG - only the first command is executed

### DISCUSSION THREAD

**Discussion by thermseekr** [arendst/Tasmota#22226] [Q&A] [answered]
Hello all,

Edit: I'm compiling 14.2.0 env:tasmota32 in Platformio.

I'm trying to run a few comands in USER_BACKLOG after setting a custom template and selecting module 0, but only the first command is being executed. Here is the relevant part of user_config_override.h:

```
#define USER_TEMPLATE   "{\"NAME\":\"ESP-24V-16CH-V2\",\"GPIO\":[0,3200,0,3232,0,0,0,0,5536,10240,10241,0,0,0,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,608,640,0,0,0,0,0,0],\"FLAG\":0,\"BASE\":1}"

#undef MODULE
#define MODULE USER_MODULE

#define USER_BACKLOG "ethipaddress 10.112.66.22; ethsubnetmask 255.255.255.0; ethgateway 10.112.66.254; wifi 0"
```

If I change the order of the commands, say make "wifi 0" the first one, again only this one is executed. And if I try the same commands in backlog on a terminal, it executes them all correctly:

```
00:11:42.279 CMD: backlog ethipaddress 10.112.66.22; ethsubnetmask 255.255.255.0; ethgateway 10.112.66.254; wifi 0
00:11:42.324 RSL: RESULT = {"EthIPAddress":"10.112.66.22"}
00:11:42.574 RSL: RESULT = {"EthSubnetmask":"255.255.255.0"}
00:11:42.824 RSL: RESULT = {"EthGateway":"10.112.66.254"}
00:11:43.072 RSL: RESULT = {"Wifi":"OFF","Mode":""}
```
What am I doing wrong, can someone see it?

Thanks.
Tales Maschio

**Comment by thermseekr** [comment]
Hi, any ideas?

**Comment by Th0maz** [comment]
Just guessing...
Try without the spaces between commands

**Reply by thermseekr** [reply]
Tried that, thanks anyway 👍🏻

**Comment by tsmoneta** [comment, marked as answer]
The documentation says it's only executed on the first boot after a reset (no variables defined in flash). If you've already changed a configuration setting, that's not the case. Perhaps a rule with 'on system#boot' would be more appropriate to your use case. 

![image](https://github.com/user-attachments/assets/2a9ac58f-6a2a-43a8-9717-bdbe8b2fcfe9)

**Reply by thermseekr** [reply]
Ah! I missed that it's only executed on first run. Maybe I got confused by the fact that the first instruction in the backlog is always executed. Or at least it looks like. Will need to check again.

Thanks for the help.

Tales

**Comment by clscrzyc** [comment]
hey,bro,I have also encountered the same problem. Have you solved it? Do you have any good solutions? Is this a bug in Tasmota?

**Reply by sfromis** [reply]
The solution was already in the marked answer. There was no issue at all. Maybe what you ask about is just not "the same".

Beware that commands in that one-time initial boot will fail if they try to config a feature not yet enabled by pin config or other commands. Restart is often needed before such take effect, and thereby out of scope for `USER_BACKLOG`
